docs/docsite/rst/user_guide/complex_data_manipulation.rst

```diff
@@ -0,0 +1,214 @@
+            "bridged",
+            "routed",
+            "bridged"
+        ]
+    }
+
+
+.. _find_mount_point:
+
+Find mount point
+----------------
+In this case, we want to find the mount point for a given path across our machines, since we already collect mount facts, we can use the following:
+
+.. code-block:: YAML+Jinja
+ :caption: Use selectattr to filter mounts into list I can then sort and select the last from
+ :emphasize-lines: 7
+
+   - hosts: all
+     gather_facts: True
+     vars:
+        path: /var/lib/cache
+     tasks:
+     - name: The mount point for {{path}}, found using the Ansible mount facts, [-1] is the same as the 'last' filter
+       debug: msg="{{(ansible_facts.mounts|selectattr('mount', 'in', path)|list|sort(attribute='mount'))[-1]['mount']}}"
+
+
+
+Omit elements from a list
+-------------------------
+
+The special ``omit`` variable ONLY works with module options, but we can still use it in other ways as an identifier to tailor a list of elements:
+
+.. code-block:: YAML+Jinja
+ :caption: Inline list filtering when feeding a module option
+ :emphasize-lines: 3, 7
+
+    - name: enable a list of windows features, by name
+      win_feature:
+        name: "{{ namestuff|reject('equalto', omit)|list }}"
+      state: absent
+      vars:
+        namestuff:
+          - "{{ (fs_installed_smb_v1|default(False))|ternary(omit, 'FS-SMB1') }}"
+          - "foo"
+          - "bar"
+
+
+Another way is to avoid adding elements to the list in the first place, so you can just use it directly:
+
+.. code-block:: YAML+Jinja
+ :caption: using set_fact in a loop to increment a list conditionally
+ :emphasize-lines: 3, 4, 6
+
+    - name: build unique list with some items conditionally omittted
+      set_fact:
+         namestuff: ' {{ (namestuff|default([]))|union([item]) }}'
+      when: item != omit
+      loop:
+          - "{{ (fs_installed_smb_v1|default(False))|ternary(omit, 'FS-SMB1') }}"
+          - "foo"
+          - "bar"
+
+
+.. _complex_type_transfomations:
+
+Complex Type transformations
+=============================
+
+Jinja provides filters for simple data type tranformations (int, bool, etc), but when you want to tranform data structures things are not as easy.
+You can use the loops and list comprehensions as shown above to help, also other filters and lookups can be chained and leverage to achieve these comlex transformations.
+
+
+.. _create_dictionary_from_list:
+
+Create dictionary from list
+---------------------------
+
+In most languages it is easy to create a dictionary (aka map/associative array/hash/etc) from a list of pairs, in Ansible there are a couple of ways to do it and the best one for you might depend on the source of your data.
+
+
+These example produces ``{"a": "b", "c": "d"}``
+
+.. code-block:: YAML+Jinja
+ :caption: Simple list to dict by assuming the list is [key, value , key, value, ...]
+
+  vars:
+      single_list: [ 'a', 'b', 'c', 'd' ]
+      mydict: "{{ dict(singlelist) }}"
```

Comment:
There's a missing underscore in the variable name. It should be:
`mydict: "{{ dict(single_list) }}"`

Reply:
You speak about a list of pairs, but that would be `[('a', 'b'), ('c', 'd')]`.

`dict(['a', 'b', 'c', 'd'])` is an error in Python:

> ValueError: dictionary update sequence element #0 has length 1; 2 is required